pilot/control/monitor.py

```diff
@@ -17,16 +16,16 @@
 
 
 def control(queues, traces, args):
-
     while not args.graceful_stop.is_set():
-        time.sleep(30 * 60)
+        if args.graceful_stop.wait(30 * 60) or args.graceful_stop.is_set():  # 'or' added for 2.6 compatibility reasons
+            break
         run_checks(args)
         send_heartbeat()
 
 
 def check_local_space_limit():
```

Comment:
By the way: what does this function check? Not sure what this free_space_limit is. Usually its a good practice to name functions that return a bool after a yes/no question (e.g. has_free_space or is_space_limit_exceeded). 
When this function gets executed now (without waiting the first 30 minutes), du[2] is much bigger than the other expression for me. So the function returns false an thus sets graceful_stop. This way pilot shuts down immediately. Is this expected? Maybe because of not waiting the 30 minutes?